### PR TITLE
layers: Improve perf for sparse buffer copies validation

### PR DESCRIPTION
**Pull request by arno-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8224

Stress test:
For 10,000 copies, went down from +15 minutes (stopped measured) down to 60 ms
For 4000, went down from +6 minutes to 10ms

For the stress test, the source and destination buffer are bound to only one VkDeviceMemory, I need to make another one where they are bound to multiples to see if perf is bad

~~Need to port this logic to the non sparse buffer case, will make another PR as users do not appear to be blocked on that~~ done, see `BufferCopiesValidationStressTest2 `

### CODE REVIEW

File: layers/state_tracker/device_memory_state.cpp [hunk 1]
```diff
@@ -192,6 +198,59 @@ BoundMemoryRange vvl::BindableSparseMemoryTracker::GetBoundMemoryRange(const Mem
     return mem_ranges;
 }
 
+BoundRanges vvl::BindableSparseMemoryTracker::GetBoundRanges(const BufferRange &ranges_bounds,
+                                                             const std::vector<BufferRange> &buffer_ranges) const {
+    BoundRanges memory_to_bound_ranges_map;
+    auto guard = ReadLockGuard{binding_lock_};
+
+    auto bound_memory_ranges = binding_map_.bounds(ranges_bounds);
```
Comment (arno-lunarg) [inline layers/state_tracker/device_memory_state.cpp:206]: I think most of the gains comes from the fact that `bounds` will be called only twice (once for the source ranges, once for the destination ranges), when it used to be called for every pair of range

File: layers/state_tracker/device_memory_state.h [hunk 2]
```diff
@@ -149,6 +155,10 @@ class BindableLinearMemoryTracker : public BindableMemoryTracker {
                     VkDeviceSize resource_offset, VkDeviceSize size) override;
 
     BoundMemoryRange GetBoundMemoryRange(const MemoryRange &range) const override;
+    BoundRanges GetBoundRanges(const BufferRange &ranges_bounds, const std::vector<BufferRange> &ranges) const override {
```
Comment (arno-lunarg) [inline layers/state_tracker/device_memory_state.h:158]: ~~GetBoundRanges has for now has no reason to be called on non sparse memory~~ Well it will soon

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 3]
```diff
@@ -1819,48 +1821,78 @@ void CoreChecks::PostCallRecordCmdCopyImage2(VkCommandBuffer commandBuffer, cons
 template <typename RegionType>
 void CoreChecks::RecordCmdCopyBuffer(VkCommandBuffer commandBuffer, VkBuffer srcBuffer, VkBuffer dstBuffer, uint32_t regionCount,
                                      const RegionType *pRegions, const Location &loc) {
+    VVL_ZoneScoped;
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:1824]: we plan to keep this here?

File: layers/profiling/profiling.h [hunk 4]
```diff
@@ -22,19 +22,49 @@
...
+// Define CPU zones
 #define VVL_ZoneScoped ZoneScoped
 #define VVL_ZoneScopedN(name) ZoneScopedN(name)
 #define VVL_TracyCZone(zone_name, active) TracyCZone(zone_name, active)
 #define VVL_TracyCZoneEnd(zone_name) TracyCZoneEnd(zone_name)
 #define VVL_TracyCFrameMark TracyCFrameMark
+
+// Print messages
 #define VVL_TracyMessage TracyMessage
+#define VVL_TracyMessageStream(message)                \
```
Comment (spencer-lunarg) [inline layers/profiling/profiling.h:34]: this should be in a separate commit IMO

File: layers/core_checks/cc_copy_blit_resolve.cpp (hunk 3 above)
Comment (arno-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1824]: Could remove yes

File: layers/profiling/profiling.h (hunk 4 above)
Comment (arno-lunarg) [reply layers/profiling/profiling.h:34]: ok!

File: layers/core_checks/cc_copy_blit_resolve.cpp (hunk 3 above)
Comment (arno-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1824]: can remove

Comment (spencer-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1824]: thanks, same as below, I really don't want to start having random `VVL_ZoneScopedN` scatter everywhere

**Comment by arno-lunarg** [comment]
On my AMD laptot, new test `PositiveSparseBuffer, BufferCopiesValidationStressTest2` is causing a `vk_timeout` when waiting for the fence after submitting memory bindings, but this timeout is hit immediately, not after the supplied delay of 10 seconds. No validation error. Can't see what is wrong with the test

File: layers/state_tracker/device_memory_state.h (hunk 2 above)
Comment (spencer-lunarg) [reply layers/state_tracker/device_memory_state.h:158]: maybe worth adding a `todo` comment here incase you don't get back to fixing this soon and someone just sees this random function that shouldn't exists and get confused

**Review by spencer-lunarg — APPROVED** [review]
